src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -607,21 +552,16 @@ internal override async ValueTask ShutdownCompleted(CancellationToken cancellati
             using CancellationTokenRegistration registration = cancellationToken.UnsafeRegister(static (s, token) =>
             {
                 var state = (State)s!;
-                bool shouldComplete = false;
                 lock (state)
                 {
                     if (state.ShutdownState == ShutdownState.None)
                     {
                         state.ShutdownState = ShutdownState.Canceled;
-                        shouldComplete = true;
                     }
                 }
 
-                if (shouldComplete)
-                {
-                    state.ShutdownWriteCompletionSource.SetException(
-                        ExceptionDispatchInfo.SetCurrentStackTrace(new OperationCanceledException("Wait for shutdown was canceled", token)));
-                }
+                // TODO: Don't we need to cancel the waiter here?
```

Comment:
We do; it was most probably a copy-paste bug https://github.com/dotnet/runtime/issues/55706#issuecomment-880617582

Reply:
https://github.com/dotnet/runtime/pull/55993

Reply:
Thanks for fixing. I merged with your change and re-pushed.